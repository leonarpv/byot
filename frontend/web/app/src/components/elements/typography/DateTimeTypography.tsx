import * as React from 'react';
import {makeStyles, Theme, Typography, TypographyProps} from '@material-ui/core';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';

interface Props extends WithStyles<typeof styles>, TypographyProps {}

const styles = (theme: Theme) => ({
  root: {
    color: theme.palette.grey[500],
  },
});
const useStyles = makeStyles(styles);

export const DateTimeTypography: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <Typography variant="caption" {...props} classes={{...props.classes, root: styles.root}}>
      {props.children}
    </Typography>
  );
};
