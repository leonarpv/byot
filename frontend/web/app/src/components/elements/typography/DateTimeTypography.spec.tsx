import * as React from 'react';
import {render} from '@testing-library/react';
import {DateTimeTypography} from './DateTimeTypography';

describe('<DateTimeTypography/>', () => {
  it('should render', () => {
    const {container} = render(<DateTimeTypography />);
    expect(container).toMatchSnapshot();
  });
});
