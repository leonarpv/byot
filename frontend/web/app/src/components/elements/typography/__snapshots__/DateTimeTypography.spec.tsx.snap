// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DateTimeTypography/> should render 1`] = `
<div>
  <span
    class="MuiTypography-root makeStyles-root-1 MuiTypography-caption"
  />
</div>
`;
